apis/Google.Devtools.Aspnet/Google.Devtools.Aspnet/CloudErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,182 @@
+        /// </summary>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static CloudErrorReportingExceptionLogger Create(string projectId, string serviceName, string version)
+        {
+            return new CloudErrorReportingExceptionLogger(
+                ReportErrorsServiceClient.Create(), projectId, serviceName, version);
+        }
+
+        private CloudErrorReportingExceptionLogger(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) : base()
+        {
+            _client = GaxPreconditions.CheckNotNull(client, nameof(client));
+            _projectId = ReportErrorsServiceClient.FormatProjectName(
+                GaxPreconditions.CheckNotNull(projectId, nameof(projectId)));
+            _serviceContext = new ServiceContext
+            {
+                Service = GaxPreconditions.CheckNotNull(serviceName, nameof(serviceName)),
+                Version = GaxPreconditions.CheckNotNull(version, nameof(version)),
+            };
+        }
+
+        public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
+        {
+            ReportedErrorEvent errorEvent = CreateReportRequest(context);
+            return _client.ReportErrorEventAsync(_projectId, errorEvent);
+        }
+
+        public override void Log(ExceptionLoggerContext context)
+        {
+            ReportedErrorEvent errorEvent = CreateReportRequest(context);
+            _client.ReportErrorEvent(_projectId, errorEvent);
+        }
+
+        public override bool ShouldLog(ExceptionLoggerContext context)
+        {
+            return context?.Exception != null;
+        }
+
+        /// <summary>
+        /// Gets information about the HTTP request and response when the exception occured 
+        /// and populates a <see cref="HttpRequestContext"/> object.
+        /// </summary>
+        private HttpRequestContext CreateHttpRequestContext(ExceptionLoggerContext exceptionLoggerContext)
+        {
+            ExceptionContext exceptionContext = exceptionLoggerContext.ExceptionContext;
+            HttpRequestMessage requestMessage = exceptionContext?.Request;
+            HttpResponseMessage responseMessage = exceptionContext?.Response;
+
+            HttpRequestContext requestContext = new HttpRequestContext()
+            {
+                Method = requestMessage?.Method?.ToString(),
+                Url = requestMessage?.RequestUri?.ToString(),
+                UserAgent = requestMessage?.Headers?.UserAgent?.ToString(),
+            };
+
+            if (responseMessage?.StatusCode != null)
+            {
+                requestContext.ResponseStatusCode = (int)responseMessage.StatusCode;
```

Comment:
You could do this unconditionally in the object initializer, as

```csharp
ResponseStatusCode = (int) (responseMessage?.StatusCode ?? 0);
```

(Because 0 will be the default value in the proto anyway.)

At that point, you don't even need the local variable for `responseMessage` - it can just be `return new HttpRequestContext { ... }`. (Note that you don't need the `()` when you're writing an object initializer.)